### PR TITLE
drivers: video: ov5640: Fix out-of-bound coverity issue

### LINKED ISSUE #81982: [Coverity CID: 434539] Out-of-bounds read in drivers/video/ov5640.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/b1def7145fd/drivers/video/ov5640.c

Category: Memory - illegal accesses
Function: `ov5640_enum_frmival`
Component: Drivers
CID: [434539](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=434539)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/b1def7145fd/drivers/video/ov5640.c

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=434539

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERRUN). ([CWE-125](http://cwe.mitre.org/data/definitions/125.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by trunghieulenxp**
Fix #81982

### CODE REVIEW

**Review by ngphibang — CHANGES_REQUESTED** [review]
Need to fix upper bound too:

fie->index >= ARRAY_SIZE(ov5640_frame_rates)

File: drivers/video/ov5640.c [hunk 1]
```diff
@@ -904,7 +904,8 @@ static int ov5640_enum_frmival(const struct device *dev, enum video_endpoint_id
 		}
 	}
 
-	if (i == ARRAY_SIZE(modes) || fie->index > ARRAY_SIZE(ov5640_frame_rates) ||
+	if (i == ARRAY_SIZE(modes) || fie->index < 0 ||
+	    fie->index > ARRAY_SIZE(ov5640_frame_rates) ||
```
[the remark this answers is not in the document]
Comment (trunghieulenxp) [reply drivers/video/ov5640.c]: Fixed it. Thanks

**Review by ngphibang — APPROVED** [review]

**Review by loicpoulain — DISMISSED** [review]

File: drivers/video/ov5640.c [hunk 2]
```diff
@@ -904,7 +904,8 @@ static int ov5640_enum_frmival(const struct device *dev, enum video_endpoint_id
 		}
 	}
 
-	if (i == ARRAY_SIZE(modes) || fie->index > ARRAY_SIZE(ov5640_frame_rates) ||
+	if (i == ARRAY_SIZE(modes) || fie->index < 0 ||
+	    fie->index >= ARRAY_SIZE(ov5640_frame_rates) ||
```
Comment (ngphibang) [inline drivers/video/ov5640.c:908]: You should remove `fie->index < 0`, just do `fie->index >= ARRAY_SIZE(ov5640_frame_rates)` as `fie->index` is an unsigned value. If you compare it with 0, you will run into another coverity issue as [here](https://github.com/zephyrproject-rtos/zephyr/issues/81936)

**Review by ngphibang — CHANGES_REQUESTED** [review]
Sorry, I just rechecked, `fie->index < 0` needs to be dropped as it is an unsigned value unless we will run into another coverity issue as https://github.com/zephyrproject-rtos/zephyr/issues/81936

**Review by josuah — APPROVED** [review]

**Review by ngphibang — APPROVED** [review]

**Comment by ngphibang** [comment]
Commit message should be updated too "Fix the out of bound buffer read by checking the lower-bound index
in the ov5640_enum_frmival function.". Now it's not lower bound but upper-bound :-) or simply just "Add check for array out of bound". But this is minor.

**Pull request merged**